docs/events.md

```diff
@@ -0,0 +1,144 @@
+# About the mbed OS event loop
+
+One of the optional mbed OS features is an event loop mechanism that can be used to defer the execution of code in a different context. In particular, a common uses of an event loop is to postpone the execution of a code sequence from an interrupt handler to an user context. This is useful because of the specific constraints of code that runs in an interrupt handler:
+
+- the execution of certain functions (notably some functions in the C library) is not safe.
+- various RTOS objects and functions can't be used from an interrupt context.
+- as a general rule, the code needs to finish as fast as possible, to allow other interrupts to be handled.
+
+The event loop offers a solution to these issues in the form of an API that can be used to defer execution of code from the interrupt context to the user context. More generally, the event loop can be used anywhere in a program (not necessarily in an interrupt handler) to defer code execution to a different context.
+
+# Overview of the mbed OS event loop
+
+An event loop has two main components:
+
+1. an **event queue**, used to store events. In mbed OS, *events* are pointers to functions (and optionally function arguments).
+2. an **event loop** that extracts events from the queue and executes them.
+
+The mbed OS event queue is implemented by the [mbed-events library](http://github.com/ARMmbed/mbed-os/tree/master/events). It's a good idea to go through the [README of mbed-events](https://github.com/ARMmbed/mbed-os/blob/master/events/README.md), as it shows how to use the event queue.
+
+The event loop must be created and started manually. The simplest way to achieve that is to create a `Thread` and run the event queue's `dispatch` method in the thread:
+
+```
+#include "mbed.h"
+#include "mbed_events.h"
+
+using events::EventQueue;
```

Comment:
> it is a completely optional component at the moment.

This is also true of the nsapi, and that library is also exposed through mbed.h (for better or worse). It's even under the features directory.

If it's a conditional inclusion (such as rtos and nsapi) than users can opt-out by overriding the `MBED_CONF_EVENTS_PRESENT` config option.

I don't have a strong feeling if you think we should keep the mbed-events as opt-in, just concerned about consistency.